plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/launcher/DataflowPipelineLaunchDelegate.java

```diff
@@ -206,14 +206,14 @@ private void setLoginCredential(ILaunchConfigurationWorkingCopy workingCopy, Str
         throw new CoreException(new Status(Status.ERROR, DataflowCorePlugin.PLUGIN_ID, message));
       }
 
-      Credential credential = loginService.getCredential(accountEmail);
-      if (credential == null) {
+      Optional<Credential> credential = googleApiFactory.getCredential();
+      if (!credential.isPresent()) {
         String message = "The Google account saved for this launch is not logged in.";
         throw new CoreException(new Status(Status.ERROR, DataflowCorePlugin.PLUGIN_ID, message));
```

Comment:
Alternative syntax sugar:
```java
Credential credential = googleApiFactory.getCredential()
    .orElseThrow(() -> new CoreException(new Status(
        Status.ERROR, 
        DataflowCorePlugin.PLUGIN_ID, 
        "The Google account saved for this launch is not logged in."))
```
Then you don't need the explicit `get()`, as it's unwrapped for you.